src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/api/Load.kt

```diff
@@ -0,0 +1,61 @@
+package it.krzeminski.snakeyaml.engine.kmp.api
+
+import it.krzeminski.snakeyaml.engine.kmp.constructor.BaseConstructor
+import it.krzeminski.snakeyaml.engine.kmp.constructor.StandardConstructor
+import okio.Source
+
+/**
+ * Common way to load object instance(s).
+ *
+ * This class is not thread-safe. Which means that all the methods of the same instance can be
+ * called only by one thread. It is better to create an instance for every YAML stream. The
+ * instance is stateful. Only one of the 'load' methods may be called, and it may be called only once.
+ *
+ * @param settings configuration
+ * @param constructor custom YAML constructor
+ */
+expect class Load(
+    settings: LoadSettings = LoadSettings.builder().build(),
+    constructor: BaseConstructor = StandardConstructor(settings),
+) {
+    /**
+     * Parse a YAML document and create an instance of an object.
+     *
+     * The string must be encoded as UTF-8.
+     *
+     * @param string YAML data to load from (BOM must not be present)
+     * @return parsed instance
+     * @throws it.krzeminski.snakeyaml.engine.kmp.exceptions.YamlEngineException if the YAML is not valid
+     */
+    fun loadOne(string: String): Any?
+
+    /**
+     * Parse a YAML document and create an instance of an object.
+     *
+     * @param source YAML data to load
+     * @return parsed instance
+     * @throws it.krzeminski.snakeyaml.engine.kmp.exceptions.YamlEngineException if the YAML is not valid
+     */
+    fun loadOne(source: Source): Any?
```

Comment:
suggestion: I think the `okio` dependency should be added to `commonMain` dependencies as `api` instead of `implementation` since it is exposed in public API

Reply:
Ah yes, good point. I think it'd be best not to expose Okio as a required dependency. Ideally the official library https://github.com/Kotlin/kotlinx-io/ will become more fully-featured, and we can migrate from Okio. So I'll make these functions internal.